### PR TITLE
EPMRPP-33453 Dashboard Widgets Launches duration

### PR DESCRIPTION
**Pull request by norbert-csaba-herczeg-epam**

### CODE REVIEW

File: app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.jsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      isConfigReady: true,
+    });
+  };
+
+  getCoords = ({ pageX, pageY }) => {
+    this.x = pageX;
+    this.y = pageY;
+  };
+
+  resizeChart = () => {
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.jsx]: Can it be handled by the `C3Chart` component itself?

File: app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+import { LaunchesDurationChart } from './launchesDurationChart';
+import README from './README.md';
+
+const mockNode = document.createElement('node');
+const mockObserver = {
+  subscribe: () => {},
+  unsubscribe: () => {},
+};
+const widgetData1 = Object.freeze({
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.stories.jsx]: widgetData

File: app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.jsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+import { DURATION } from './constants';
+import styles from './launchesDurationChart.scss';
+
+const cx = classNames.bind(styles);
+
+@injectIntl
+export class LaunchesDurationChart extends Component {
+  static propTypes = {
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.jsx]: `preview`

File: app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.jsx (hunk 1 above)
Comment (norbert-csaba-herczeg-epam) [reply app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.jsx]: @DiscoElevator The resize algorithm is the same as it is in the reference Chart component (`LaunchesComparisonChart`). It reacts to the wrapping components size changing, and resizes the C3Chart explicitly. I believe this solution should be kept unless we see a recurring pattern in the handling of resize events.

File: app/src/common/constants/localization.js [hunk 4]
```diff
@@ -49,4 +49,8 @@ export const COMMON_LOCALE_KEYS = defineMessages({
     id: 'Common.closeModalWarning',
     defaultMessage: 'You have to save changes or cancel them before closing the window',
   },
+  SECONDS: {
+    id: 'Common.dateTime.seconds',
```
Comment (DiscoElevator) [inline app/src/common/constants/localization.js]: Common.seconds

File: app/localization/translated/be.json [hunk 5]
```diff
@@ -78,6 +78,7 @@
   "Common.cancel": "Адмяніць",
   "Common.closeModalWarning": "Вы павінны захаваць змены або адмяніць іх перад закрыццём вокны",
   "Common.confirm": "Пацвердзіць",
+  "Common.dateTime.seconds": "seconds",
```
Comment (DiscoElevator) [inline app/localization/translated/be.json]: Please add translation from https://github.com/reportportal/service-ui/blob/localization_by/src/main/resources/public/js/src/localizations/be-BY.js#L1997

File: app/localization/translated/ru.json [hunk 6]
```diff
@@ -79,6 +79,7 @@
   "Common.cancel": "Отменить",
   "Common.closeModalWarning": "Вы должны сохранить изменения или отменить их перед закрытием окна",
   "Common.confirm": "Подтвердить",
+  "Common.dateTime.seconds": "seconds",
```
Comment (DiscoElevator) [inline app/localization/translated/ru.json]: Please add translation from https://github.com/reportportal/service-ui/blob/localization_by/src/main/resources/public/js/src/localizations/ru-RU.js#L2049

File: app/localization/translated/be.json [hunk 7]
```diff
@@ -78,7 +78,7 @@
   "Common.cancel": "Адмяніць",
   "Common.closeModalWarning": "Вы павінны захаваць змены або адмяніць іх перад закрыццём вокны",
   "Common.confirm": "Пацвердзіць",
-  "Common.dateTime.seconds": "seconds",
+  "Common.dateTime.seconds": "секунды/секунд",
```
Comment (DiscoElevator) [inline app/localization/translated/be.json]: The key should also be updated: `Common.seconds`

File: app/localization/translated/ru.json [hunk 8]
```diff
@@ -79,7 +79,7 @@
   "Common.cancel": "Отменить",
   "Common.closeModalWarning": "Вы должны сохранить изменения или отменить их перед закрытием окна",
   "Common.confirm": "Подтвердить",
-  "Common.dateTime.seconds": "seconds",
+  "Common.dateTime.seconds": "секунды/секунд",
```
Comment (DiscoElevator) [inline app/localization/translated/ru.json]: `Common.seconds`

**Comment by norbert-csaba-herczeg-epam** [comment]
@DiscoElevator updated.

**Comment by norbert-csaba-herczeg-epam** [comment]
@DiscoElevator Updated with v5 compatible data handling.

**Pull request merged**